classes/local/factor/object_factor_base.php

```diff
@@ -215,4 +215,35 @@ public function login_form_definition_after_data($mform) {
     public function login_form_validation($data) {
         return array();
     }
+
+    /**
+     * Returns true if factor class has factor records that might be deleted.
+     * It means that user can delete factor record from their profile.
+     *
+     * Override in child class if necessary.
+     *
+     * @return bool
+     */
+    public function has_delete() {
+        return false;
+    }
+
+    /**
+     * Marks factor record as disabled.
```

Comment:
I'm not keen on the inconsistent language, we revoke, but then call a method delete, which toggles a field called disabled. Can you make it revoke / revoked / can_revoke / revoke_user_factor  everywhere